### PR TITLE
Refactor flask backend

### LINKED ISSUE #171: Refactor flask-backend structure
**Is your feature request related to a problem? Please describe.**
It is required to provide a backend structure to flask-backend for robust development.

**Describe the solution you'd like**
Present structure could be improved by providing environment variable through .env file, creating a config.py that would provide separate configuration according to environment like Production, Testing or development and using enabling migrations for database versions.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by sachinsom93**
# Description
Done following changes
Commit 1:
       Corrected some path issue in apiutils.py
Commit 2:
       Refactored flask-structure via factory pattern.
Commit 3:
       Updated flask-backend Readme according to new changes
 
Fixes #171 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
postman

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: flask-backend/migrations/versions/cb09cbe62f0c_initial_migrate.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+"""initial migrate
```
Comment (shivanshuraj1333) [inline flask-backend/migrations/versions/cb09cbe62f0c_initial_migrate.py:1]: Do we need to commit this file?
If not, we can add such files to gitignore.

**Review by shivanshuraj1333 — COMMENTED** [review]
take a look

File: flask-backend/migrations/versions/cb09cbe62f0c_initial_migrate.py (hunk 1 above)
Comment (sachinsom93) [reply flask-backend/migrations/versions/cb09cbe62f0c_initial_migrate.py:1]: @shivanshu1333, yes we need to keep this file publicly, as this file shows initial versions and commits of database classes and changes.

Comment (sachinsom93) [reply flask-backend/migrations/versions/cb09cbe62f0c_initial_migrate.py:1]: With help of this file other developers can upgrade and degrade to database commit versions according to their interest :)

**Review by shivanshuraj1333 — APPROVED** [review]
LGTM

**Pull request merged**